packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -282,27 +329,57 @@ export default function vercelServerless({
 							includeFiles,
 							excludeFiles,
 							maxDuration,
+							isr,
 						});
 						routeDefinitions.push({
 							src: route.pattern.source,
 							dest: func,
 						});
 					}
 				} else {
-					await createFunctionFolder({
-						functionName: NODE_PATH,
-						runtime,
-						entry: new URL(_serverEntry, _buildTempFolder),
-						config: _config,
-						logger,
-						NTF_CACHE,
-						includeFiles,
-						excludeFiles,
-						maxDuration,
-					});
+					if (isr) {
+						await createFunctionFolder({
+							functionName: '_isr',
+							runtime,
+							entry: new URL(_serverEntry, _buildTempFolder),
+							config: _config,
+							logger,
+							NTF_CACHE,
+							includeFiles,
+							excludeFiles,
+							maxDuration,
+							isr,
+						});
+					}
+					if (isr === false || (typeof isr === "object" && isr.exclude)) {
```

Comment:
Avoid errors when `isr === null` since `typeof null === 'object'`. Technically not a valid type, but I've been bitten enough to watch for this 😓 

```suggestion
					if (isr === false || (isr && typeof isr === "object" && isr.exclude)) {
```